### DISCUSSION TITLE
Is it possible to reference WinForms from Win UI 3 project?

### DISCUSSION THREAD

**Discussion by nikolayvpavlov** [microsoft/microsoft-ui-xaml#7757] [Q&A] [answered]
The title says it all.

I tried by adding  `<useWindowsForms>true</useWindowsForms>` in the proj file, but this cased the project to stop building on line `InitializeComponents` in the Window's partial class.  Obviously the above breaks the build and the compiler cannot file the generated file.

So, is there a way to reference WinForms and use some of its classes in a WinUI 3 project?

**Comment by castorix** [comment, marked as answer]
Yes, by adding a reference to System.Windows.Forms.dll (+ other DLLs like System.Drawing.Common.dll or System.Resources.Extensions.dll, depending on what you need)
(from _C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App_ + version)

**Comment by KWodarczyk** [comment]
In my case, the project builds but as soon as i open page there references winforms .dll i get the following error.
[The system cannot find the file specified. File name: 'System.Windows.Forms,]

or 
An attempt was made to load a program with an incorrect format. (0x8007000B)

**Reply by gautambjain** [reply]
May be related to 32bit / 64bit?
